Source/mhd/hlld.f90

```diff
@@ -1,441 +1,459 @@
+    integer           :: UMAGN, UMAGP1, UMAGP2
+    integer           :: i,j,k
+
+    type (eos_t) :: eos_state
+
+    ! `n` here is the normal
+    ! `p` are the perpendicular
+
+    if (dir .eq. 1) then
+       QMAGN  = QMAGX
+       QMAGP1 = QMAGY
+       QMAGP2 = QMAGZ
+       QVELN  = QU
+       QVELP1 = QV
+       QVELP2 = QW
+       UMN    = UMX
+       UMP1   = UMY
+       UMP2   = UMZ
+       UMAGN  = UMAGX
+       UMAGP1 = UMAGY
+       UMAGP2 = UMAGZ
+
+    else if (dir .eq. 2) then
+       QMAGN  = QMAGY
+       QMAGP1 = QMAGZ
+       QMAGP2 = QMAGX
+       QVELN  = QV
+       QVELP1 = QW
+       QVELP2 = QU
+       UMN    = UMY
+       UMP1   = UMZ
+       UMP2   = UMX
+       UMAGN  = UMAGY
+       UMAGP1 = UMAGZ
+       UMAGP2 = UMAGX
+
+    else if (dir .eq. 3) then
+       QMAGN  = QMAGZ
+       QMAGP1 = QMAGX
+       QMAGP2 = QMAGY
+       QVELN  = QW
+       QVELP1 = QU
+       QVELP2 = QV
+       UMN    = UMZ
+       UMP1   = UMX
+       UMP2   = UMY
+       UMAGN  = UMAGZ
+       UMAGP1 = UMAGX
+       UMAGP2 = UMAGY
+    end if
+
+    do k = work_lo(3), work_hi(3)
+       do j = work_lo(2), work_hi(2)
+          do i = work_lo(1), work_hi(1)
+
+             if (dir .eq. 1) then
+                qL(:) = qleft(i-1,j,k,:)
+             else if (dir .eq. 2) then
+                qL(:) = qleft(i,j-1,k,:)
+             else if (dir .eq. 3) then
+                qL(:) = qleft(i,j,k-1,:)
+             end if
+
+             qR(:) = qright(i,j,k,:)
+
+             !flx(i,j,k,:) = 0.d0
+             !FL  = 0.d0; FR = 0.d0; UsL = 0.d0; UsR = 0.d0; FsL = 0.d0; FsR = 0.d0; UssL = 0.d0; UssR = 0.d0; FssL = 0.d0; FssR = 0.d0
+
+             ! check to not go lower than small_dens and small_press
+             qL(QRHO) = max(small_dens, qL(QRHO))
+             qR(QRHO) = max(small_dens, qR(QRHO))
+             qL(QPRES) = max(small_pres, qL(QPRES))
+             qR(QPRES) = max(small_pres, qR(QPRES))
+
+             call PToC(qL, uL, gam1_L)
+
+             eintL = UL(UEINT)/UL(URHO)
+
+             ! Compute the fluxes.  Here use total p not just p_g eq.11 in Miniati
+             ! Also note that Miniati has an overall sign error in the B fluxes
+             BL2 = dot_product(qL(QMAGX:QMAGZ), qL(QMAGX:QMAGZ))
+
+             FL(URHO) = qL(QRHO)*qL(QVELN)
+             FL(UMN) = qL(QRHO)*qL(QVELN)**2 + (qL(QPRES) + 0.5d0*BL2) - qL(QMAGN)**2
+             FL(UMP1) = qL(QRHO)*qL(QVELN)*qL(QVELP1) - qL(QMAGN)*qL(QMAGP1)
+             FL(UMP2) = qL(QRHO)*qL(QVELN)*qL(QVELP2) - qL(QMAGN)*qL(QMAGP2)
+             FL(UEDEN) = qL(QVELN)*(UL(UEDEN) + (qL(QPRES) + 0.5d0*BL2)) - &
+                  qL(QMAGN)*dot_product(qL(QMAGX:QMAGZ), qL(QU:QW))
+             FL(UMAGN) = 0.d0
+             FL(UMAGP1) = qL(QVELN)*qL(QMAGP1) - qL(QVELP1)*qL(QMAGN)
+             FL(UMAGP2) = qL(QVELN)*qL(QMAGP2) - qL(QVELP2)*qL(QMAGN)
+             FL(UFS:UFS+nspec-1) = qL(QVELN) * UL(UFS:UFS+nspec-1)
+             FL(UEINT) = qL(QRHO)*qL(QVELN)*eintL
```

Comment:
Since rho is the same in the primitive and conserved states, we could simplify this to qL(QVELN) * UL(UEINT) and remove the eintL and eintR variables

Reply:
changed... but note that eintL/R were used in one other place in the code